### PR TITLE
fix(images): omit port when api.port is empty or undefined

### LINKED ISSUE #3963: fix(images): setImages builds malformed URL when api.port is empty
## Context

Since the Gravatar → `UserAvatarComponent` refactor (PR #3927, commit `a5f4e9ab`, 2026-04-09), user avatars are rendered via `setImages(config.api, user.avatar, size * 2, null)` — routed through the shared `setImages` helper in `src/lib/plugins/images.js`.

On downstream projects deployed over HTTPS with an empty `api.port` (the standard prod setup — port 443 implicit), avatars are broken: the rendered URL contains a stray colon with no port, e.g. `https://trawl.me:/api/uploads/images/abc.jpg`, which browsers reject.

Repro confirmed on trawl_vue (built with `DEVKIT_VUE_api_port=` in CI). Affects every user with an uploaded avatar on the scraps list and anywhere else `UserAvatarComponent` is used. Users without uploaded avatars see the initials fallback and are unaffected.

## Root cause

`src/lib/plugins/images.js:17` unconditionally concatenates `:${api.port}`:

```js
return \`${api.protocol}://${api.host}:${api.port}/api/uploads/images/${name}\`;
```

When `api.port === ''`, this produces `https://host:/api/...` — malformed.

Other URL builders in the codebase already handle this correctly — e.g. `trawl_vue/src/modules/scraps/utils/screenshotUrl.js:12-13`:

```js
const port = config.api.port ? \`:${config.api.port}\` : '';
return \`${config.api.protocol}://${config.api.host}${port}/${config.api.base}/uploads/${screenshot}\`;
```

The `setImages` plugin was never updated with this pattern, but the bug was latent because pre-refactor avatars went through Gravatar (remote URL, not `setImages`) — the refactor exposed it.

## Scope

**Bug affects all downstream projects built with empty `api.port`** (i.e., HTTPS prod on standard 443). Since the refactor shipped 2026-04-09, every downstream that ran `/update-stack` since then has broken avatars.

### 1. `src/lib/plugins/images.js`

Replace the single return line with conditional port handling:

```js
app.config.globalProperties.setImages = (api, file, size, operation) => {
  const base = file.split('.');
  if (base.length !== 2) return file;
  let [name] = base;
  if (size) name = \`${name}-${size}\`;
  if (operation) name = \`${name}-${operation}\`;
  name = \`${name}.${base[1]}\`;
  const port = api.port ? \`:${api.port}\` : '';
  return \`${api.protocol}://${api.host}${port}/api/uploads/images/${name}\`;
};
```

Single-line change in the return, plus one `port` local. No signature change, no API break.

### 2. `src/lib/plugins/tests/images.unit.tests.js`

Add a test case for empty port:

```js
it('omits port when api.port is empty', () => {
  const api = { protocol: 'https', host: 'trawl.me', port: '' };
  expect(setImages(api, 'photo.jpg', null, null))
    .toBe('https://trawl.me/api/uploads/images/photo.jpg');
});

it('omits port when api.port is undefined', () => {
  const api = { protocol: 'https', host: 'trawl.me' };
  expect(setImages(api, 'photo.jpg', null, null))
    .toBe('https://trawl.me/api/uploads/images/photo.jpg');
});
```

Existing tests (which pass `port: '3000'`) remain valid and unchanged.

### 3. No other files touched

- `UserAvatarComponent`: no change — it passes `config.api` as-is, the fix is transparent.
- `MIGRATIONS.md`: no entry needed (bugfix, not a breaking change).

## Risks

None. Purely additive / defensive — URLs with a port continue to render identically.

## Validation

- `/verify` on the stack (lint + unit + audit)
- Manual: deploy a downstream (trawl_vue first via `/update-stack`), check `https://trawl.me` scraps list — avatars should load instead of 404.

## Downstream rollout

Once merged, each affected downstream needs `/update-stack` to pick up the fix: `trawl_vue`, `pierreb_vue`, `comes_vue`, `montaine_vue` (and any other HTTPS-deployed project).

## Estimated cost

~10 min real work. Single PR, ~5 LOC in source + ~10 LOC in tests.

### PR DESCRIPTION
**Pull request by PierreBrisorgueil**
## Summary

- `setImages` was unconditionally concatenating `:${api.port}`, producing malformed URLs like `https://host:/api/...` on HTTPS deployments where port is empty (standard prod on port 443).
- Broke user avatars everywhere `UserAvatarComponent` is used after the Gravatar refactor (commit a5f4e9ab, #3927), since avatars now route through `setImages`.
- Adds conditional port handling matching the pattern already used in downstream `screenshotUrl.js`. Purely additive/defensive — URLs with a port render identically.

## Test plan

- [x] Lint clean
- [x] Unit tests: 996 passed (73 files), including 2 new cases (empty port string + undefined port)
- [x] Coverage stable (98.99% stmts, 99.51% lines) — no thresholds touched
- [x] Existing tests with `port: '3000'` unchanged and still passing

Fixes #3963

## Release Notes

* **Bug Fixes**
  * Fixed image URL formation to correctly handle cases where the API port is not specified or empty.

* **Tests**
  * Added tests to verify proper URL construction when API port is not provided.

### CODE REVIEW

File: src/lib/plugins/images.js [hunk 1]
```diff
@@ -14,7 +14,8 @@ export default {
       if (size) name = `${name}-${size}`;
       if (operation) name = `${name}-${operation}`;
       name = `${name}.${base[1]}`;
-      return `${api.protocol}://${api.host}:${api.port}/api/uploads/images/${name}`;
+      const port = api.port ? `:${api.port}` : '';
+      return `${api.protocol}://${api.host}${port}/api/uploads/images/${name}`;
```
[the remark this answers is not in the document]
Comment (PierreBrisorgueil) [reply src/lib/plugins/images.js:32]: Fixed in ad103fa6: added JSDoc headers for both `install` and `setImages` with description, `@param` for each argument, and `@returns`, per ERRORS.md 2026-02-22.
